packer/plugin-getter/plugins.go

```diff
@@ -845,51 +845,58 @@ func (pr *Requirement) InstallLatest(opts InstallOptions) (*Installation, error)
 							return nil, errs
 						}
 
-						outputFile, err := os.OpenFile(outputFileName, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
+						tmpBinFileName := filepath.Join(os.TempDir(), expectedBinaryFilename)
 						if err != nil {
-							err := fmt.Errorf("failed to create %s: %w", outputFileName, err)
+							err = fmt.Errorf("could not create temporary file to download plugin: %w", err)
+							errs = multierror.Append(errs, err)
+							return nil, errs
+						}
+						defer func() {
+							tmpBinPath := tmpBinFileName
+							os.Remove(tmpBinPath)
+						}()
+
+						tmpOutputFile, err := os.OpenFile(tmpBinFileName, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
+						if err != nil {
+							err := fmt.Errorf("failed to create %s: %w", tmpBinFileName, err)
 							errs = multierror.Append(errs, err)
 							return nil, errs
 						}
-						if _, err := io.Copy(outputFile, copyFrom); err != nil {
+						if _, err := io.Copy(tmpOutputFile, copyFrom); err != nil {
 							err := fmt.Errorf("extract file: %w", err)
 							errs = multierror.Append(errs, err)
-							os.Remove(outputFileName)
 							return nil, errs
 						}
 
-						if _, err := outputFile.Seek(0, 0); err != nil {
+						if _, err := tmpOutputFile.Seek(0, 0); err != nil {
 							err := fmt.Errorf("Error seeking beginning of binary file for checksumming: %w", err)
 							errs = multierror.Append(errs, err)
 							log.Printf("[WARNING] %v, ignoring", err)
 						}
 
-						outputFileCS, err := checksum.Checksummer.Sum(outputFile)
+						outputFileCS, err := checksum.Checksummer.Sum(tmpOutputFile)
 						if err != nil {
 							err := fmt.Errorf("failed to checksum binary file: %s", err)
 							errs = multierror.Append(errs, err)
 							log.Printf("[WARNING] %v, ignoring", err)
 						}
-						outputFile.Close()
+						tmpOutputFile.Close()
 
-						desc, err := GetPluginDescription(outputFileName)
+						desc, err := GetPluginDescription(tmpBinFileName)
 						if err != nil {
-							err := fmt.Errorf("failed to describe plugin binary %q: %s", outputFileName, err)
+							err := fmt.Errorf("failed to describe plugin binary %q: %s", tmpBinFileName, err)
 							errs = multierror.Append(errs, err)
-							os.Remove(outputFileName)
 							continue
 						}
 						descVersion, err := goversion.NewSemver(desc.Version)
 						if err != nil {
 							err := fmt.Errorf("invalid self-reported version %q: %s", desc.Version, err)
 							errs = multierror.Append(errs, err)
-							os.Remove(outputFileName)
 							continue
 						}
 						if descVersion.Core().Compare(version.Core()) != 0 {
 							err := fmt.Errorf("binary reported version (%q) is different from the expected %q, skipping", desc.Version, version.String())
 							errs = multierror.Append(errs, err)
-							os.Remove(outputFileName)
 							continue
```

Comment:
Like the PrereleaseInstallError I think having an error to denote the we can should try the next version  would help wrap this logic in a function so that we could then use error.Is to determine if we continue or return immediately .

Reply:
Yeah I agree, we should maybe introduce a concept of a `ContinuableError` somewhere in there.

Reply:
✅